src/main/java/com/powsybl/caseserver/service/CaseService.java

```diff
@@ -0,0 +1,170 @@
+        createCaseMetadataEntity(newCaseUuid, withExpiration, withIndexation, null, null, caseMetadataRepository);
+    }
+
+    default List<CaseInfos> getMetadata(List<UUID> ids) {
+        List<CaseInfos> cases = new ArrayList<>();
+        ids.forEach(caseUuid -> {
+            CaseInfos caseInfos = getCaseInfos(caseUuid);
+            if (Objects.nonNull(caseInfos)) {
+                cases.add(caseInfos);
+            }
+        });
+        return cases;
+    }
+
+    default Importer getImporterOrThrowsException(Path caseFile, ComputationManager computationManager) {
+        DataSource dataSource = DataSource.fromPath(caseFile);
+        Importer importer = Importer.find(dataSource, computationManager);
+        if (importer == null) {
+            throw CaseException.createFileNotImportable(caseFile);
+        }
+        return importer;
+    }
+
+    default byte[] createZipFile(Collection<String> names, MemDataSource dataSource) throws IOException {
+        try (var outputStream = new ByteArrayOutputStream();
+             var zipOutputStream = new ZipOutputStream(outputStream)) {
+            for (String name : names) {
+                zipOutputStream.putNextEntry(new ZipEntry(name));
+                zipOutputStream.write(dataSource.getData(name));
+                zipOutputStream.closeEntry();
+            }
+            return outputStream.toByteArray();
+        }
+    }
+
+    default Optional<ExportCaseInfos> exportCase(UUID caseUuid, String format, Map<String, Object> formatParameters) throws IOException {
+        if (!Exporter.getFormats().contains(format)) {
+            throw CaseException.createUnsupportedFormat(format);
+        }
+
+        var optionalNetwork = loadNetwork(caseUuid);
+        if (optionalNetwork.isPresent()) {
+            var network = optionalNetwork.get();
+            var memDataSource = new MemDataSource();
+            Properties exportProperties = null;
+            if (formatParameters != null) {
+                exportProperties = new Properties();
+                exportProperties.putAll(formatParameters);
+            }
+
+            network.write(format, exportProperties, memDataSource);
+
+            var listNames = memDataSource.listNames(".*");
+            String networkName = DataSourceUtil.getBaseName(getCaseName(caseUuid));
+            byte[] networkData;
+            if (listNames.size() == 1) {
+                String extension = listNames.iterator().next();
+                networkName += extension;
+                networkData = memDataSource.getData(extension);
+            } else {
+                networkName += ".zip";
+                networkData = createZipFile(listNames, memDataSource);
+            }
+            return Optional.of(new ExportCaseInfos(networkName, networkData));
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    default List<CaseInfos> getCasesToReindex(CaseMetadataRepository caseMetadataRepository) {
```

Comment:
remove method entirely from CaseService, Call the one from casemetadatarepository in supervisioncontroller ?

Reply:
It uses caseService.getCases(). So better to keep it here